cmd/src/batch_exec.go

```diff
@@ -290,33 +256,13 @@ func convertWorkspace(w batcheslib.WorkspacesExecutionInput) service.RepoWorkspa
 			Commit:      graphql.Target{OID: w.Branch.Target.OID},
 			FileMatches: fileMatches,
 		},
-		Path:               w.Path,
-		Steps:              w.Steps,
-		OnlyFetchWorkspace: w.OnlyFetchWorkspace,
+		Path:                  w.Path,
+		Steps:                 w.Steps,
+		OnlyFetchWorkspace:    w.OnlyFetchWorkspace,
+		BatchChangeAttributes: &w.BatchChangeAttributes,
+		CachedStepResultFound: w.CachedStepResultFound,
+		CachedStepResult:      w.CachedStepResult,
```

Comment:
we now construct the task here already, so we don't need to read the file from disk again and might risk cache key mismatches.